code/Hieu_L/python/classes_lab.py

```diff
@@ -0,0 +1,56 @@
+import random
+class bank_account:
+    def __init__(self, name, balance, account_number):
+        self.name = name
+        self.balance = balance
+        self.account_number = account_number
+        # self.account_number = random.randint(0,9)
+    def __repr__(self):
+        return print(f"account number: {self.account_number} \n name: {self.name} \n account number: {self.account_number}")
+
+    def deposit(self, amount):
+        # amount = input('enter deposit amount: ')
+        self.balance += amount
+        return self.balance
+    
+    def withdrawal(self, amount):
+        # amount = input('enter withdrawal amount: ')
+        self.balance -= amount
+        return self.balance
+
+class rectangle:
+    def __init__(self, length, width):
+        self.length = length
+        self.width = width
+    
+    def area(self):
+        area = self.length * self.width 
+        return area
+    
+    def perimeter(self):
+        perimeter = (2 * self.length) + (2 * self.width)
+        return perimeter
+
+    def __repr__(self):
+        area = area()
+        return f"rectangle dimensions: \n length: {self.length} \n width: {self.width} \n area: {area}"
+
+class parallelepiped(rectangle):
+    def __init__(self, height):
```

Comment:
Inheritance needs needs correct syntax. You're missing the super method invocation when creating the parallelepiped class. Here's an example:

class Rabbit(Animal):
    def __init__(self, legs, color, speed): ##adding the new attribute speed
        super().__init__(legs, color) ## referencing old attributes legs,color
        self.speed = speed